packages/features/insights/server/routing-events.ts

```diff
@@ -0,0 +1,459 @@
+      form: formsWhereCondition,
+    };
+
+    // Get total forms count
+    const totalForms = await prisma.app_RoutingForms_Form.count({
+      where: formsWhereCondition,
+    });
+
+    // Get total responses
+    const totalResponses = await prisma.app_RoutingForms_FormResponse.count({
+      where: responsesWhereCondition,
+    });
+
+    // Get responses without booking
+    const responsesWithoutBooking = await prisma.app_RoutingForms_FormResponse.count({
+      where: {
+        ...responsesWhereCondition,
+        routedToBookingUid: null,
+      },
+    });
+
+    return {
+      created: totalForms,
+      total_responses: totalResponses,
+      total_responses_without_booking: responsesWithoutBooking,
+      total_responses_with_booking: totalResponses - responsesWithoutBooking,
+    };
+  }
+
+  static async getRoutingFormsForFilters({
+    teamId,
+    isAll,
+    organizationId,
+  }: {
+    teamId?: number;
+    isAll: boolean;
+    organizationId?: number | undefined;
+    routingFormId?: string | undefined;
+  }) {
+    const formsWhereCondition = await this.getWhereForTeamOrAllTeams({ teamId, isAll, organizationId });
+    return await prisma.app_RoutingForms_Form.findMany({
+      where: formsWhereCondition,
+      select: {
+        id: true,
+        name: true,
+        _count: {
+          select: {
+            responses: true,
+          },
+        },
+      },
+    });
+  }
+
+  static async getRoutingFormPaginatedResponses({
+    teamId,
+    startDate,
+    endDate,
+    isAll,
+    organizationId,
+    routingFormId,
+    cursor,
+    limit,
+    userId,
+    fieldFilter,
+    bookingStatus,
+  }: RoutingFormInsightsFilter & { cursor?: number; limit?: number }) {
+    const formsTeamWhereCondition = await this.getWhereForTeamOrAllTeams({
+      teamId,
+      isAll,
+      organizationId,
+      routingFormId,
+    });
+
+    const responsesWhereCondition: Prisma.App_RoutingForms_FormResponseWhereInput = {
+      ...(startDate &&
+        endDate && {
+          createdAt: {
+            gte: dayjs(startDate).startOf("day").toDate(),
+            lte: dayjs(endDate).endOf("day").toDate(),
+          },
+        }),
+      ...(userId || bookingStatus
+        ? {
+            ...(bookingStatus === "NO_BOOKING"
+              ? { routedToBooking: null }
+              : {
+                  routedToBooking: {
+                    ...(userId && { userId }),
+                    ...(bookingStatus && { status: bookingStatus }),
+                  },
+                }),
+          }
+        : {}),
+      ...(fieldFilter && {
+        response: {
+          path: [fieldFilter.fieldId, "value"],
+          array_contains: [fieldFilter.optionId],
+        },
+      }),
+      form: formsTeamWhereCondition,
+    };
+
+    const totalResponsePromise = prisma.app_RoutingForms_FormResponse.count({
+      where: responsesWhereCondition,
+    });
+
+    const responsesPromise = prisma.app_RoutingForms_FormResponse.findMany({
```

Comment:
Looks like a lot but this data is pretty small when fetched in chunks of 10 at a time.